Engine/include/Decorators.h

```diff
@@ -0,0 +1,60 @@
+//
+// Decorators.h
+// VGD Engine
+//
+// Created by Benjamin Arnold on 29 Aug 2015
+//
+
+/*! \file Decorators.h
+* @brief Function decorators for clarity.
+*/
+
+#pragma once
+
+#define OUT /*!< This pointer/reference will be modified as an extra return value */
+#define OPT /*!< This pointer is optional (may be null) */
+
+#define CALLEE_DELETE /*!< Function called will delete this resource */
+#define CALLER_DELETE /*!< Function caller must manage this resource */
+
+#define VORB_INTERNAL /*!< Internal function not meant for an end-user */
+
+#define UNIT_SPACE(SPACE_TYPE) /*!< The decorated value is found in <SPACE_TYPE> space */
+
+/*! @brief Encloses M in an unnamed struct so it can be used inside a union.
+*/
+#define UNIONIZE(M) struct { M; }
```

Comment:
"GCC does support it, and so does Visual C++. Windows API headers make use of this feature by default" SO we should be ok